doc/source/whatsnew/v1.1.0.rst

```diff
@@ -701,6 +701,8 @@ Deprecations
   instead (:issue:`34191`).
 - The ``squeeze`` keyword in the ``groupby`` function is deprecated and will be removed in a future version (:issue:`32380`)
 
+- A timedelta passed a number string without a defined unit is deprecated (:issue:`12136`)
```

Comment:
can you reference ``:class:`Timedelta` `` and ``pandas.to_timedelta`` here

Reply:
why are we deprecating this? i would simply remove it as its wrong

Reply:
@jreback By "we simply DO not allow a unit when the input is a string"? Do you refer to the unit inside the string or the unit parameter? This change behaviors are like this:

- Allows `to_timedelta("1s")`, used as example in https://pandas.pydata.org/docs/user_guide/timedeltas.html#to-timedelta
- Allows  `to_timedelta("1", unit="s")`, fixes bug #12136
- Disallows `to_timedelta("1s", unit="s")`
- Warns `to_timedelta("1")`

I simply merged @Sup3rGeo's PR #23025 without changing any behaviors. Please let me know if you want something to be changed.

Reply:
we spent a long time on the PR but was ultimately not merged.because of the way it was structured. Please re-read those comments. We want to handle this at a much higher level, if you pass a unit then your input *must* be numeric, there is not other option. I don't think PR needs to be very complicated at all.